.buildkite/x-pack/pipeline.xpack.otel.yml

```diff
@@ -0,0 +1,138 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+name: "beats-xpack-otel"
+
+env:
+  AWS_ARM_INSTANCE_TYPE: "m6g.xlarge"
+  AWS_IMAGE_UBUNTU_ARM_64: "platform-ingest-beats-ubuntu-2204-aarch64"
+
+  IMAGE_UBUNTU_X86_64: "family/platform-ingest-beats-ubuntu-2204"
+
+  IMAGE_BEATS_WITH_HOOKS_LATEST: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci-with-hooks:latest"
+
+  GCP_DEFAULT_MACHINE_TYPE: "c2d-highcpu-8"
+
+  #Deps
+  ASDF_MAGE_VERSION: 1.15.0
+
+  # Unit tests
+  RACE_DETECTOR: "true"
+  TEST_COVERAGE: "true"
+
+steps:
+  - group: "check"
+    key: "x-pack-otel-check"
+
+    steps:
+      - label: "x-pack/otel: Run check"
+        command: |
+          set -eo pipefail
+          make -C x-pack/otel check
+          make check-no-changes
+        retry:
+          automatic:
+            - limit: 1
+        agents:
+          image: "${IMAGE_BEATS_WITH_HOOKS_LATEST}"
+          cpu: "4000m"
+          memory: "8Gi"
+          useCustomGlobalHooks: true
+        notify:
+          - github_commit_status:
+              context: "x-pack/otel: check"
+
+      - label: "x-pack/otel: Run pre-commit"
+        command: "pre-commit run --all-files"
+        agents:
+          image: "${IMAGE_BEATS_WITH_HOOKS_LATEST}"
+          memory: "2Gi"
+          useCustomGlobalHooks: true
+        notify:
+          - github_commit_status:
+              context: "x-pack/otel: pre-commit"
+
+  - wait: ~
+    # with PRs, we want to run mandatory tests only if check/update step succeed
+    # for other cases, e.g. merge commits, we want to run mundatory test (and publish) independently of other tests
+    # this allows building DRA artifacts even if there is flakiness in check/update step
+    if: build.env("BUILDKITE_PULL_REQUEST") != "false"
+    depends_on: "x-pack-otel-check"
+
+  - group: "Mandatory Tests"
+    key: "x-pack-otel-mandatory-tests"
+
+    steps:
+      - label: ":ubuntu: x-pack/otel: Ubuntu x86_64 Unit Tests"
+        key: "x-pack-otel-mandatory-linux-unit-test"
+        command: |
+          cd x-pack/otel
+          mage goUnitTest
+        retry:
+          automatic:
+            - limit: 1
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
+        artifact_paths:
+          - "x-pack/otel/build/*.xml"
+          - "x-pack/otel/build/*.json"
+        plugins:
+          - test-collector#v1.10.2:
+              files: "x-pack/otel/build/TEST-*.xml"
+              format: "junit"
+              branches: "main"
+              debug: true
+        notify:
+          - github_commit_status:
+              context: "x-pack/otel: Ubuntu x86_64 Unit Tests"
+
+      - label: ":ubuntu: x-pack/otel: Ubuntu x86_64 Unit Tests with requirefips build tag"
```

Comment:
A few more comments, if `"x-pack-otel-mandatory-linux-unit-test-FIPS-tag"` and `"x-pack-otel-mandatory-linux-unit-test"` only differ in the env variable, can you use the buildkite matrix syntax instead?

https://buildkite.com/docs/pipelines/configure/workflows/build-matrix